iroh-net/src/endpoint.rs

```diff
@@ -858,9 +858,7 @@ impl Endpoint {
         tracing::debug!("Closing connections");
         endpoint.close(error_code, reason);
         endpoint.wait_idle().await;
-        // In case this is the last clone of `Endpoint`, dropping the `quinn::Endpoint` will
-        // make it more likely that the underlying socket is not polled by quinn anymore after this
-        drop(endpoint);
```

Comment:
@divagant-martian does it make sense to store the `quinn::Endpoint` in an `Option` so that it can be taken out and dropped?  That would also mean that regardless how many clones there are of `iroh_net::Endpoint` they will all lose the `quinn::Endpoint` when closed, which can be used as a very good indicator that the endpoint is closed.

Reply:
`Arc<Option<EndPoint>>`?

Reply:
having to put that into a mutex would not be great.  it already has mutexes inside it.

Reply:
ok, we should not try and drop the quinn endpoint here.